### DISCUSSION TITLE
Flask-socketio Sticky Sessions

### DISCUSSION THREAD

**Discussion by aktoak** [miguelgrinberg/Flask-SocketIO#1759] [Q&A]
i want to clarify something about the use of sticky session.
i'd be glad if you can tell me if these statements are correct

**one worker:**
1) if i only use 1 worker i don't need stick sessions at all.

**multiple works:**
1) if i used flask-socketio with `manage_session=True`, and because flask-socketio uses memory to store client session then i do need stick sessions.
2) if i used flask-socketio with  `manage_session=False`, and use Flask-Sessions with redis for example to manage client sessions
but allow http polling for websockets connections then i still need sticky sessions for http polling.
3) if i used flask-socketio with  `manage_session=False`, and use Flask-Sessions with redis for example to manage client sessions
and permit only websocket connections  (without long polling) then i also **don't** need sticky session?

**Comment by miguelgrinberg** [comment]
No, you have some things wrong. You seem to be confusing user sessions with sticky sessions and these are completely different things. The user session, is where you store client specific data. The sticky sessions are a configuration setting for your load balancer. I would rewrite your analysis as follows:

1. If you have one worker, then you don't need a load balancer, so speaking about sticky sessions in this context makes no sense.
2. If you have multiple workers, then your load balancer needs to be configured with sticky sessions _in the normal case_.
3. (Exception to 2) If your clients connect directly with the WebSocket protocol (instead of doing the normal connection with long-polling and then upgrading to WebSocket), then sticky sessionsi n the load balancer aren't necessary and can be omitted. Not using sticky sessions in the load balancer basically breaks long-polling.
